### PR TITLE
fix: Fix for Ansys Lab upload/download methods

### PR DESCRIPTION
**Pull request by hpohekar**
Closes https://github.com/ansys/pyfluent/issues/2591

```
import ansys.fluent.core as pyfluent
solver_session = pyfluent.launch_fluent(product_version="24.2.0")
solver_session.file.read(file_type="mesh", file_name="mixing_elbow.msh.h5")
Fast-loading "/ansys_inc/v241/fluent/fluent24.1.0/addons/afd/lib/hdfio.bin"
Done.

Reading from fluent-3ddp-241-xwiuxy27is2s5aisj5lfpspyg05ep8dt5suwh-7bkjt:"/work/mixing_elbow.msh.h5" in NODE0 mode ...
  Reading mesh ...
       17517 cells,     1 cell zone  ...
          17517 mixed cells,  zone id: 3451
       89857 faces,     7 face zones ...
           2169 polygonal wall faces,  zone id: 34
            271 polygonal wall faces,  zone id: 33
            155 polygonal pressure-outlet faces,  zone id: 32
            152 polygonal velocity-inlet faces,  zone id: 31
             54 polygonal velocity-inlet faces,  zone id: 30
           1993 polygonal symmetry faces,  zone id: 29
          85063 mixed interior faces,  zone id: 3453
       64786 nodes,     3 node zones ...

Building...
     mesh
     materials,
     interface,
     domains,
     zones,
	Skipping thread 20 of domain 1 (not referenced by grid).
	Skipping thread 21 of domain 1 (not referenced by grid).
	Skipping thread 22 of domain 1 (not referenced by grid).
	Skipping thread 23 of domain 1 (not referenced by grid).
	Skipping thread 24 of domain 1 (not referenced by grid).
	Skipping thread 25 of domain 1 (not referenced by grid).
	Skipping thread 26 of domain 1 (not referenced by grid).
	Skipping thread 27 of domain 1 (not referenced by grid).
	Skipping thread 28 of domain 1 (not referenced by grid).
	wall-elbow
	wall-inlet
	outlet
	cold-inlet
	hot-inlet
	symmetry-xyplane
	interior--elbow-fluid
	elbow-fluid
     parallel,
Done.
Mesh is now scaled to meters.
solver_session.exit()
```

Progressbar Integration

image

### CODE REVIEW

**Review by prmukherj — APPROVED** [review]

**Review by seanpearsonuk — APPROVED** [review]

**Review by mkundu1 — APPROVED** [review]

**Review by ochernuk — APPROVED** [review]

File: src/ansys/fluent/core/session.py [hunk 1]
```diff
@@ -329,6 +339,16 @@ def download(self, file_name: str, local_directory: Optional[str] = "."):
         local_directory : str, optional
             Local destination directory. The default is the current working directory.
         """
+        warnings.warn(
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session.py:342]: @hpohekar You have copied an unintended method name on the first line of the warning. You could factor out the warning call into a private method and pass in the name of the method.

Comment (hpohekar) [reply src/ansys/fluent/core/session.py:342]: Right, Thank you.

Comment (hpohekar) [reply src/ansys/fluent/core/session.py]: Done. Thank you.

File: src/ansys/fluent/core/session.py [hunk 2]
```diff
@@ -308,16 +308,31 @@ def force_exit(self) -> None:
         data."""
         self._fluent_connection.force_exit()
 
-    def upload(self, file_name: str):
+    def _warning(self, method_name: str) -> str:
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session.py:311]:
```suggestion
    def _file_transfer_api_warning(self, method_name: str) -> str:
```

File: src/ansys/fluent/core/session.py [hunk 3]
```diff
@@ -308,16 +308,31 @@ def force_exit(self) -> None:
...
         """Upload a file to the server.
 
         Parameters
         ----------
         file_name : str
             Name of the local file to upload to the server.
+        remote_file_name : str, optional
+            remote file name, by default None
         """
+        warnings.warn(self._warning("upload()"), UserWarning)
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session.py:333]: Given that the intention was to factor out all duplication, more could be delegated to the called function.

**Comment by seanpearsonuk** [comment]
The internal implementation for PIM has the `file_exist()` capability. Should we expose it along with `upload()` and `download()` at the session level? It would perhaps be called `file_exists_on_remote()`?

**Comment by hpohekar** [comment]
> The internal implementation for PIM has the `file_exist()` capability. Should we expose it along with `upload()` and `download()` at the session level? It would perhaps be called `file_exists_on_remote()`?

@seanpearsonuk Yes, but it is particular to PIM-based file transfer service. it is not available for gRPC-based file transfer service.

image

https://filetransfer.tools.docs.pyansys.com/version/dev/api/_autosummary/ansys.tools.filetransfer.Client.html#ansys.tools.filetransfer.Client

I think we can implement `file_exist()` for us (I'm not sure now but we can try) or we can request this method to PIM team.

File: src/ansys/fluent/core/session.py (hunk 2 above)
Comment (hpohekar) [reply src/ansys/fluent/core/session.py]: Done. Thank you.

File: src/ansys/fluent/core/session.py (hunk 3 above)
Comment (hpohekar) [reply src/ansys/fluent/core/session.py]: Done. Thank you.

**Comment by seanpearsonuk** [comment]
> > The internal implementation for PIM has the `file_exist()` capability. Should we expose it along with `upload()` and `download()` at the session level? It would perhaps be called `file_exists_on_remote()`?
> 
> @seanpearsonuk Yes, but it is particular to PIM-based file transfer service. it is not available for gRPC-based file transfer service.
> 
> image
> 
> https://filetransfer.tools.docs.pyansys.com/version/dev/api/_autosummary/ansys.tools.filetransfer.Client.html#ansys.tools.filetransfer.Client
> 
> I think we can implement `file_exist()` for us (I'm not sure now but we can try) or we can request this method to file-transfer-tool team.

@hpohekar 
Yes, I was wondering about the gRPC-based version.
If the implementation does not support it, we have a number of options to deal with that at runtime. Maybe it shouldn't stop us exposing it if it is possible for some backends?

**Comment by hpohekar** [comment]
> > > The internal implementation for PIM has the `file_exist()` capability. Should we expose it along with `upload()` and `download()` at the session level? It would perhaps be called `file_exists_on_remote()`?
> > 
> > 
> > @seanpearsonuk Yes, but it is particular to PIM-based file transfer service. it is not available for gRPC-based file transfer service.
> > image
> > https://filetransfer.tools.docs.pyansys.com/version/dev/api/_autosummary/ansys.tools.filetransfer.Client.html#ansys.tools.filetransfer.Client
> > I think we can implement `file_exist()` for us (I'm not sure now but we can try) or we can request this method to file-transfer-tool team.
> 
> @hpohekar Yes, I was wondering about the gRPC-based version. If the implementation does not support it, we have a number of options to deal with that at runtime. Maybe it shouldn't stop us exposing it if it is possible for some backends?

Yes, sure. I will look into it.

**Pull request merged**